File: pset6/caesar.py
# import modules
import sys
import cs50


# main
def main():
    # ensure right usage
    if len(sys.argv) != 2:
        print("Usage: python caesar.py k")
        exit(1)

    # get k
    k = int(sys.argv[1])

    # get plain text
    print("plaintext:  ", end="")
    p = cs50.get_string()

    # print cipher text
    print("ciphertext: ", end="")
    for i in range(len(p)):
        c = p[i]
        if c.isalpha():
            if c.isupper():
                print("{}".format(chr((ord(c) - 65 + k) % 26 + 65)), end="")
            else:
                print("{}".format(chr((ord(c) - 97 + k) % 26 + 97)), end="")
        else:
            print("{}".format(c), end="")
    print()

    # done
    exit(0)


# redirect to main
if __name__ == "__main__":
    main()

File: pset2/caesar/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <stdlib.h>


int main(int argc, string argv[]){
    if(argc != 2){
        printf("Usage: ./caesar k\n");
        exit(1);
    }
    
    int k = atoi(argv[1]);
    printf("plaintext:  ");
    string p = get_string();
    
    printf("ciphertext: ");
    for(int i = 0; i < strlen(p); i++){
        if(isalpha(p[i])){
            if(isupper(p[i])){
                printf("%c", (p[i] - 65 + k) % 26 + 65);
            }else{
                printf("%c", (p[i] - 97 + k) % 26 + 97);
            }
        }else{
            printf("%c", p[i]);
        }
    }
    
    printf("\n");
}